api_app/analyzers_manager/observable_analyzers/vt3_sandbox.py

```diff
@@ -0,0 +1,90 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import time
+import logging
+
+import requests
+
+from api_app.exceptions import AnalyzerRunException
+from api_app.analyzers_manager.file_analyzers import vt3_scan
+from api_app.analyzers_manager.classes import ObservableAnalyzer
+
+from tests.mock_utils import patch, if_mock_connections, MockResponse
+
+logger = logging.getLogger(__name__)
+
+vt_base = "http://www.virustotal.com/api/v3/"
+
+
+class VirusTotalv3Sandbox(ObservableAnalyzer):
+    def set_params(self, params):
+        self.__api_key = self._secrets["api_key_name"]
+        # max no. of tries when polling for result
+        self.max_tries = params.get("max_tries", 10)
+        # interval b/w HTTP requests when polling
+        self.poll_distance = 30
+
+    def run(self):
+        return vt_get_behavior_summary(
+            self.__api_key,
+            self.observable_name,
+            self.observable_classification,
+            self._params,
+            self.job_id,
+            max_tries=self.max_tries,
+            poll_distance=self.poll_distance,
+        )
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockResponse({}, 200),
+                ),
+            )
+        ]
+        return super()._monkeypatch(patches=patches)
+
+
+def vt_get_behavior_summary(
+    api_key,
+    observable_name,
+    obs_clfn,
+    config_params,
+    job_id,
+    max_tries=100,
+    poll_distance=5,
+):
+    headers = {"x-apikey": api_key}
+    uri = f"files/{observable_name}/behaviour_summary"
+
+    for chance in range(max_tries):
+        try:
+            logger.info(
+                f"trying VT/v3 GET n.{chance + 1} for job_id: {job_id}, "
+                f"observable {observable_name}"
+            )
+            response = requests.get(vt_base + uri, headers=headers)
+            if not response.status_code == 404:
+                response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        result = response.json()
+
+        # If file hash not found in VirusTotal Database
+        if response.status_code == 404:
```

Comment:
even if this does make sense if seen with only this perspective, in practice we cannot do that because there is already another analyzer that would perform the "active" scan if the sample is not available in the DB (`VirusTotal_v3_Get_File_And_Scan`). 

So, a user that launches "all analyzers" would ask to VT to perform an active scan 2 times. This is why other similar endpoints (like `VirusTotal_v3_Get_File`) are disabled by default. This is extremely critical when managing the available quota of an expensive service like VT.

I would not force the active scanner option.
I would add different options that can be configured:
* force the active scan if the file was not found (useful if this analyzer is run alone), default=False
* force the polling if we know that this analyzer is executed at the same time of another VT scan analyzer (case of `all analyzers` execution, the most common one), default=True. (without this flag, only a single request is done)

Thoughts?

Reply:
> force the active scan if the file was not found (useful if this analyzer is run alone), default=False

You mean similar to the `force_active_scan` config variable in the `VirusTotal_v3_Get_File_And_Scan` analyzer, right ?

Reply:
> force the polling if we know that this analyzer is executed at the same time of another VT scan analyzer (case of all analyzers execution, the most common one), default=True. (without this flag, only a single request is done)

Although this can be done easily by checking is the `self._job.run_all_analyzers` is True or not, I was thinking that maybe in that case we just don't conduct the analysis and write like "Please see the VirusTotal_v3_Get_File_And_Scan report instead" because  doesn't the other VT analyzer contain the data of this "behaviour summary" too ?

Reply:
A problem with both of these approacesh is that if a user has the other VT analyzer disabled, the analyzer will break. Maybe we should not make an analyzer dependent on another.

Reply:
the other VT analyzer does not contain this data unfortunately

Reply:
I think having them not depend on each other would be the best thing as well. If each VT analyzer can have it's own check for sending a file for analysis if required, otherwise just retrieving the result they are designed to, it would be cleaner allbeit a bit redundant.

Reply:
Another thing could be, what if there's a single analyzer that deals with sending and retrieving a report from VT and we can simple retrieve from different endpoints based on the config options. LIke default could be the default report for analysis but a `report_type = behavioral` could retrieve a behavioral report instead. (Similar to how it's done in Triage Scan currently)

Reply:
Yes, your last idea is the better one I think. We could add the behavioral report as "optional" in the already existing VT 3 analyzer

Reply:
I love that idea! just a boolean config var `include_behavioral_summary`.

Reply:
Behavioral analysis works but I'm having some trouble finding files that match to sigma rules. All the samples I tried didn't have that analysis available to them, so the default report is retrieved in place. It should work otherwise.

Reply:
Instead of a string config var like `report_type`, I think it'd be better to have a parameter like: `include_behavioral_summary`, so we will always fetch the relationship/default detail but also include the behavioral data if this param is set to True.

The problem I see with `report_type` is that user cannot request multiple types of information in a single analysis only.

@mlodic, thoughts ?

Reply:
I agree with Eshaan.

To test some samples already in VT with sigma rules try 478331245c3ad724ce0880ebd911fd54 or 2e74c9eb3caf855ea2d050b4ece174c8 and let me know if it does find anything

Reply:
Cool. Similar to behaviour one would be the `include_sigma_analyses` for sigma.

Reply:
That would be better, yeah. Although the way the analyzer is written rn, it's a bit messy. Maybe after we get the report, just before returning it, we can call on these endpoints if the config variable is set to True and append the data to the rest of the dictionary.

Reply:
yes, that's how it should be. I think we can make the default relationship values as a parameter too.

Reply:
So basically, 3 parameters:
- `include_relationships`
- `include_behaviour_summary`
- `include_sigma_analyses`

Reply:
Actually, the hashes @mlodic provided are md5 but the `sigma_analyses` API works only with sha256 hash. ([docs reference](https://developers.virustotal.com/v3.0/reference#get-sigma-analysis-id))